### PR TITLE
Trim object names when copying the missing objects list

### PR DESCRIPTION
**Pull request by Broxzier**
Before adding a name to the clipboard buffer, it checks for spaces at the end of the name, to prevent copying those.

It also moves some logic from the `mouseup` function, and moved memory deallocation within the same function where it got allocated.

Because it's a cpp file, I changed `malloc` and `SafeFree` to `new` and `delete` respectively. I wasn't sure if I should also make use of `std::string`, or leave it.

### CODE REVIEW

File: src/openrct2-ui/windows/ObjectLoadError.cpp [hunk 1]
```diff
@@ -154,29 +154,35 @@ static rct_string_id get_object_type_string(const rct_object_entry *entry)
...
+    // Something has gone wrong, this shouldn't happen.
+    // We don't want to allocate stupidly large amounts of memory for no reason
+    assert(w->no_list_items > 0 && w->no_list_items <= OBJECT_ENTRY_COUNT);
 
     // No system has a newline over 2 characters
     size_t line_sep_len = strnlen(PLATFORM_NEWLINE, 2);
     size_t buffer_len = (w->no_list_items * (8 + line_sep_len)) + 1;
-    buffer = (utf8*)malloc(buffer_len);
-    buffer[0] = '\0';
+    utf8* buffer = new utf8[buffer_len]{};
```
Comment (janisozaur) [inline src/openrct2-ui/windows/ObjectLoadError.cpp]: I haven't seen such syntax yet, what does `{}` do in this case?

Comment (Broxzier) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: It initializes the buffer to zero.

Comment (Gymnasiast) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: `utf8 * buffer`

Comment (Gymnasiast) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: ^ Can you fix above thingy?

Comment (janisozaur) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: I think clang elided this whole allocation. 👍

Comment (Gymnasiast) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: I mean the formatting.

Comment (Broxzier) [reply src/openrct2-ui/windows/ObjectLoadError.cpp]: I've updated it. Also I'm still curious to which is prefered, this buffer, or `std::string`.

File: src/openrct2/management/Award.cpp [hunk 2]
```diff
@@ -552,7 +552,10 @@ static bool award_is_deserved_best_custom_designed_rides(sint32 awardType, sint3
 }
 
 /** At least 5 colourful rides and more than half of the rides are colourful. */
-static const uint8 dazzling_ride_colours[] = {5, 14, 20, 30};
+static const uint8 DazzlingRideColours[] =
+{
+    COLOUR_BRIGHT_PURPLE, COLOUR_BRIGHT_GREEN, COLOUR_LIGHT_ORANGE, COLOUR_BRIGHT_PINK,
```
Comment (Gymnasiast) [inline src/openrct2/management/Award.cpp]: These changes do not really belong here, but you can leave them as they are, AFAIC.

Comment (Broxzier) [reply src/openrct2/management/Award.cpp]: I just noticed after pushing, one moment :P

Comment (Broxzier) [reply src/openrct2/management/Award.cpp]: I was working on those in another branch.

**Comment by Gymnasiast** [comment]
The `delete[]` counterpart of SafeFree is SafeDeleteArray.

**Comment by janisozaur** [comment]
`SafeFree*` are only relevant when the pointer might be used in a different context. Here it is fully contained within a single function, so no need to waste cycles resolving macros and resetting memory, that would probably end up optimised out anyway.

**Comment by IntelOrca** [comment]
> so no need to waste cycles resolving macros and resetting memory, that would probably end up optimised out anyway.

So why mention it?

**Comment by Gymnasiast** [comment]
Ok, can you two decide on what Broxzier should do here?

**Comment by IntelOrca** [comment]
What is he has done is perfectly fine and can be merged. If he wants to be more C++, more safe then you can change the allocation to `std::unique<utf8[]>(length);` so you do not have to delete it.

**Pull request merged**